context/app/static/js/components/workspaces/CreateWorkspaceDialog/CreateWorkspaceDialog.jsx

```diff
@@ -1,35 +1,92 @@
 import React from 'react';
 import Button from '@mui/material/Button';
+import Box from '@mui/material/Box';
 
-import CreateWorkspaceInput from 'js/components/workspaces/CreateWorkspaceInput';
 import DialogModal from 'js/shared-styles/DialogModal';
+import { useSnackbarStore } from 'js/shared-styles/snackbars';
+import { useSelectableTableStore } from 'js/shared-styles/tables/SelectableTableProvider/store';
+import WorkspaceField from 'js/components/workspaces/WorkspaceField';
 import { useCreateWorkspace } from './hooks';
 
-function CreateWorkspaceDialog({ handleCreateWorkspace, buttonComponent: ButtonComponent, defaultName, ...rest }) {
+function CreateWorkspaceDialog({
+  handleCreateWorkspace,
+  buttonComponent: ButtonComponent,
+  defaultName,
+  errorMessages,
+  protectedRows,
+  ...rest
+}) {
   const { dialogIsOpen, setDialogIsOpen, handleSubmit, handleClose, control, errors, onSubmit } = useCreateWorkspace({
     handleCreateWorkspace,
     defaultName,
   });
+
+  const protectedHubmapIds = protectedRows?.map((row) => row._source.hubmap_id).join(', ');
+  const { toastSuccess } = useSnackbarStore();
+  const { deselectRows } = useSelectableTableStore();
+
+  const removeProtectedDatasets = () => {
+    deselectRows(protectedRows.map((r) => r._id));
+    toastSuccess('Protected datasets successfully removed from selection.');
+  };
+
```

Comment:
`CreateWorkspaceDialog` is used outside of the search page on both the workspaces page and datasets pages. I don't believe this component will have access to the `SelectableTableProvider` store. We should move this to a component around `CreateWorkspaceDialog` for use on the search page.

Reply:
I've addressed this and the MetadataMenu comment accordingly; I've created a `CreateWorkspaceWithDatasetsDialog` component which wraps the `CreateWorkspaceDialog` component, performs the error checking logic necessary,